generator/intermediate_data.hpp

```diff
@@ -52,6 +52,21 @@ struct LatLonPos
 static_assert(sizeof(LatLonPos) == 16, "Invalid structure size");
 static_assert(std::is_trivially_copyable<LatLonPos>::value, "");
 
+class PointStorageWriterInterface
+{
+public:
+  virtual ~PointStorageWriterInterface() {}
+  virtual void AddPoint(uint64_t id, double lat, double lon) = 0;
+  virtual uint64_t GetNumProcessedPoints() const = 0;
```

Comment:
The original idea was that using inheritance so as not to copy a single getter is too much.
Now that you have AddPoint it's probably all right.